src/provider/provider_file_memory.c

```diff
@@ -404,8 +404,12 @@ static umf_result_t file_mmap_aligned(file_memory_provider_t *file_provider,
         "inserted a value to the map of memory mapping (addr=%p, size=%zu)",
         ptr, extended_size);
 
-    file_provider->base_mmap = ptr;
-    file_provider->size_mmap = extended_size;
+    // align the new pointer
+    uintptr_t aligned_ptr = ALIGN_UP_SAFE((uintptr_t)ptr, alignment);
+    size_t aligned_size = extended_size - (aligned_ptr - (uintptr_t)ptr);
+
+    file_provider->base_mmap = (void *)aligned_ptr;
+    file_provider->size_mmap = aligned_size;
```

Comment:
Should it be a separate PR?

Reply:
It is a separate PR: https://github.com/oneapi-src/unified-memory-framework/pull/1157
but it is required in this PR too

Reply:
@ldorau I think you mentioned the wrong PR.
PR #1157 does not contain these changes.

Reply:
Right, it is https://github.com/oneapi-src/unified-memory-framework/pull/1158

Reply:
Done

Reply:
https://github.com/oneapi-src/unified-memory-framework/pull/1158 has been merged. I have rebased this PR.